deployment/helm/values.yaml

```diff
@@ -46,7 +46,7 @@ controller:
     # controller.yaml will default to v(AppVersion)-dev
     tag:
     # pullPolicy is the Akri Controller pull policy
-    pullPolicy: Always
+    pullPolicy: IfNotPresent
```

Comment:
Should we set this only if we do not have useLatestContainers? Otherwise folks would get stuck, no? Without useLatestContainers, that should be fine, because they are picking a specific release and we are not planning to update the tags (I think). What do you think?

Reply:
Is Kubernetes not able to check whether the image digest has changed when using `ifNotPresent`? The [docs](https://kubernetes.io/docs/concepts/configuration/overview/#container-images) says you can specify a digest to make sure you always use the same version of an image. That makes me think that it could distinguish whether a "newer" latest container exists? Maybe that is too hopeful, though.

Just to get all the possibilities down on paper, if I am not mistaken, @bfjelds please correct me if I am wrong, there are 4 main containers that could be pulled (if not specifying an explicit tag). Say that the akri Helm chart was added (`helm repo add akri-helm-charts https://deislabs.github.io/akri/`) when the version was 0.1.5:
1. `akri-dev` chart: <image>:**v0.1.5-dev**
2. `akri-dev` chart with `useLatestContainers`: <image>:**latest-dev** (this could be greater than v0.1.5 if more versions have been created)
3. `akri` chart: <image>:**v0.1.5**
4. `akri` chart with `useLatestContainers`: <image>:**latest** (this could be greater than v0.1.5 if more versions have been created)

The user would need to do `helm repo update` to remove the incongruency between latest and non-latest. 

So, for 1 and 3, what you're saying makes sense, most likely those havent been updated.

Reply:
Iirc, without specifying the pullPolicy, if the tag is latest, it defaults to Always, otherwise IfNotPresent. You can override by specifying the pullPolicy. That makes me think that IfNotPresent will not refresh. Though it should be relatively easy to test this.

https://kubernetes.io/docs/concepts/containers/images/#updating-images

Reply:
Great. I can test it. Assuming the expected result you mention, maybe we remove ImagePullPolicy since it defaults to the behavior you requested

Reply:
i think, at the least, we should leave the ability to configure pullPolicy, but maybe only if a value is provided (so the default is "")